src/utils/callbackSet.ts

```diff
@@ -0,0 +1,38 @@
+///<reference path="../reference.ts" />
+
+module Plottable {
+  export module Utils {
+    export class CallbackSet<CB extends Function> {
+      private _values: CB[];
+
+      constructor() {
+        this._values = [];
+      }
+
+      public add(value: CB) {
+        if (this._values.indexOf(value) === -1) {
```

Comment:
I could call it `UniqueArray` but it's really just a `Set`. Can't get O(1) (amortized) runtime because there's no hashing. Also n is probably <100 here so I doubt this will be much of a bottleneck.